src/components/MapContainer.tsx

```diff
@@ -79,6 +82,11 @@ export const MapContainer = ({ onClick }: { onClick?: __esri.ViewImmediateClickE
 
     setSelectorOptions(selectorOptions);
 
+    mapView.current.watch(
```

Comment:
We have a hook for this we could use.

https://github.com/agrc/kitchen-sink/blob/main/packages/utilities/src/hooks/useViewLoading.js